compiler/ballerina-lang/src/main/java/io/ballerina/projects/JBallerinaBalaWriter.java

```diff
@@ -236,16 +236,24 @@ private CompilerBackend.TargetPlatform getTargetPlatform(PackageResolution pkgRe
 
         // 2) Check package has defined any platform dependency
         PackageManifest manifest = this.packageContext.project().currentPackage().manifest();
-        PackageManifest.Platform manifestPlatform = manifest.platform(this.backend.targetPlatform().code());
-        if (manifestPlatform != null &&
-                !manifestPlatform.dependencies().isEmpty() &&
-        !isPlatformDependenciesTestOnly(manifestPlatform.dependencies())) {
+        if (hasPlatformDependencies(manifest.platforms())) {
             return this.backend.targetPlatform();
         }
-
         return AnyTarget.ANY;
     }
 
+    private boolean hasPlatformDependencies(Map<String, PackageManifest.Platform> platforms) {
+        boolean platformsNotEmpty = !platforms.isEmpty();
+        boolean hasDependencies = true;
+        for (PackageManifest.Platform value: platforms.values()) {
+            if (!value.dependencies().isEmpty()) {
+                boolean isTestOnly = isPlatformDependenciesTestOnly(value.dependencies());
+                return platformsNotEmpty && hasDependencies && !isTestOnly;
```

Comment:
we can omit this `platformsNotEmpty` check and variable altogether. If it's empty won't enter the loop right?

Reply:
`hasDependencies` seem to be true always. shall we check?